app/Models/CouchNote.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Models;
+
+use Parsedown;
+use Jenssegers\Model\Model;
+use App\Helpers\CouchDbHelper;
+
+class CouchNote extends Model
+{
+    protected $hidden = [];
+    protected $guarded = [];
+
+    /**
+     * Return the markdown parsed body.
+     *
+     * @return string
+     */
+    public function getParsedBodyAttribute()
+    {
+        return (new Parsedown())->text($this->body);
+    }
+
+    /**
+     * Save the doc in the database.
+     *
+     * @return CouchNote
+     */
+    public function save($accountId = null)
+    {
+        if (! $accountId) {
+            $accountId = auth()->user()->account->id;
+        }
+        $client = CouchDbHelper::getAccountDatabase($accountId);
+
+        return $client->storeDoc($this);
+    }
+
+    /**
+     * Remove the doc from the database.
+     *
+     * @return void
+     */
+    public function delete($accountId = null)
+    {
+        if (! $accountId) {
+            $accountId = auth()->user()->account->id;
+        }
+        $client = CouchDbHelper::getAccountDatabase($accountId);
+
+        return $client->deleteDoc($this);
+    }
+
+    /**
+     * Get a note from his _id.
+     *
+     * @return CouchNote
+     */
+    public static function getOneById($accountId, $noteId)
```

Comment:
We'll probably need to do this for every class.

Perhaps we could put this method in the parent?

Reply:
Right ! I'll create a super class.